File: 문자열/9935 - 문자열 폭발.py
import sys
read = sys.stdin.readline

s = read().strip()
bomb = list(read().strip())

b_len = len(bomb)

stack = []

for i in range(len(s)) :
    stack.append(s[i])
    # 끝자리를 먼저 비교하기
    if stack[-1] == bomb[-1] :
        # 나머지가 같으면 그만큼 빼주기
        if stack[-b_len:] == bomb :
            for i in range(b_len) :
                stack.pop()

if ''.join(stack) == '' :
    print('FRULA')
else :
    print(''.join(stack))

File: 문자열/9935 - 문자열 폭발.js
const fs = require('fs');
const { exit } = require('process');
const input = fs
  .readFileSync('./문자열/9935input.txt')
  .toString()
  .trim()
  .split('\r\n');
// const input = fs.readFileSync('dev/stdin').toString().trim().split('\n');

const s = input[0];
const target = input[1];
const l = target.length;

const isCorrect = (stack) => {
  return target
    .split('')
    .every((char, index) => char === stack[stack.length - l + index]);
};

const bomb = () => {
  for (let i = 0; i < l; i++) {
    stack.pop();
  }
};

const stack = [];
let index = 0;
while (index !== s.length) {
  stack.push(s[index]);
  if (stack.length >= l && isCorrect(stack)) {
    bomb();
  }
  index++;
}

if (stack.length !== 0) {
  console.log(stack.join(''));
  process.exit();
}

console.log('FRULA');
